smalltalksrc/VMMaker-Tools/TParseNode.extension.st

```diff
@@ -1,5 +1,26 @@
 Extension { #name : #TParseNode }
 
+{ #category : #'*VMMaker-Tools' }
+TParseNode >> inspectNode: specBuilder [
+	<inspectorPresentationOrder: 910 title: 'C'>
+
+	^  SpTextPresenter new 
+		text: (String streamContents: [ : stream | 
+			| cg |
+			cg := MLVMCCodeGenerator new.
+			cg vmMaker: VMMaker new.
+			cg vmMaker vmmakerConfiguration: VMMakerConfiguration.
+			cg currentMethod: (TMethod new
+					labels: Set new;
+					definingClass: self class;
+					yourself).
+			cg pushScope: TStatementListNode new.
+			(self respondsTo: #expression)
```

Comment:
I don't think this is correct. Not all nodes answering to `#expression` are expressions.
For example, return nodes contain an expression (the expression to return) but are not expressions themselves!

Maybe we miss an `#isExpression` message?

Also, it would be valuable to see both translations (as expression and not), maybe in two tabs?
There is a way to make inspector tabs appear conditionally I think.

Reply:
Yes, you need to define a method with the same name receiving an argument. For example, check 

```smalltalk
AbstractFileReference >> inspectionContents
	<inspectorPresentationOrder: 0 title: 'Contents'> 
	...

AbstractFileReference >> inspectionContentsContext: aContext 
	aContext active: self isFile
```

We could have an equivalent with `isExpression`.

Then, if you ask yourself what nodes are expressions, you can check the ones redefining `asCASTExpressionIn:`

Reply:
I checked the active usage but I couldn't fully understood how it works, because somewhere I should set the a method selector, I tried doing something like 

```smalltalk
	self isExpression
		ifTrue: [ aContext methodSelector: #asCASTExpressionIn: ]
		ifFalse: [ aContext methodSelector: #asCASTIn: ].
```

or: 

```smalltalk
inspectNode: aContext 

	aContext active: self isExpression
```

But then if we have two tabs we wouldn't make use of the #isExpression right?

Reply:
It's by name convention. If you have a method defining an inspector tab that is called `myInspectorTab`, then you can define also `myInspectorTabContext: aContext` that wil have a configuration context as argument. And you can use that to set different properties to the tab (like visibility or other)